troubleshooting/tools/tracevol.py

```
@@ -362,28 +470,43 @@ def get_pool_name(arg, vol_id):
     stdout, stderr = out.communicate()
     if stderr is not None:
         if arg.debug:
-            print("failed to pool name %s", stderr)
+            print("failed to get the pool name %s", stderr)
         return ""
     try:
         pools = json.loads(stdout)
     except ValueError as err:
         if arg.debug:
-            print("failed to pool name %s", err)
+            print("failed to get the pool name %s", err)
         return ""
-    pool_id = vol_id.split('-')
-    if len(pool_id) < 4:
-        if arg.debug:
-            print("pood id notin proper format", pool_id)
-        return ""
-    if pool_id[3] in arg.rooknamespace:
-        pool_id = pool_id[4]
+    if isRBD:
+        pool_id = vol_id.split('-')
+        if len(pool_id) < 4:
+            if arg.debug:
+                print("pood id not in the proper format", pool_id)
```

Do you really want to just print here? Perhaps this should raise an exception?